sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/ExpectationValidation.scala

```diff
@@ -55,48 +59,41 @@ private[smartdatalake] trait ExpectationValidation { this: DataObject with Smart
    */
   def expectations: Seq[Expectation]
 
-  def setupConstraintsAndJobExpectations(df: GenericDataFrame)(implicit context: ActionPipelineContext): (GenericDataFrame, DataFrameObservation) = {
+  def setupConstraintsAndJobExpectations(df: GenericDataFrame, defaultExpectationsOnly: Boolean = false, predicateTolerant: Boolean = false, additionalJobAggExpressionColumns: Seq[GenericColumn] = Seq())(implicit context: ActionPipelineContext): (GenericDataFrame, DataFrameObservation) = {
     // add constraint validation column
-    val dfConstraints = setupConstraintsValidation(df)
+    val dfConstraints = if (defaultExpectationsOnly) df else setupConstraintsValidation(df)
```

Comment:
I dont understand why you need to make this if distinction here. When the user specifies no constraints, then contraints is empty and therfore setupConstraintsValidation does nothing anyway

Reply:
Description added